controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -446,3 +468,16 @@ func (r *DataScienceClusterReconciler) watchDataScienceClusterResources(a client
 
 	return nil
 }
+
+// argoWorkflowCRDPredicates filters the delete events to trigger reconcile when Argo Workflow CRD is deleted.
+var argoWorkflowCRDPredicates = predicate.Funcs{
+	DeleteFunc: func(e event.DeleteEvent) bool {
+		if e.Object.GetName() == datasciencepipelines.ArgoWorkflowCRD {
+			labels := e.Object.GetLabels()
+			if _, ok := labels["app.opendatahub.io/"+datasciencepipelines.ComponentName]; !ok {
```

Comment:
We could use `labels.ODH.Component(datasciencepipelines.ComponentName)`

Reply:
i have it updated in https://github.com/opendatahub-io/opendatahub-operator/pull/967/files#diff-ddd4fc79031732c6d6cbc28c4f3fae64ee035ff75633e6d19097f717f36311b6